laravel/app/Http/Controllers/MedhubController.php

```diff
@@ -67,147 +65,37 @@ public function medhubPOST($callPath, $request = '{"programID":73}')
-                    Resident::insert(["name" => $name, "email" => $osuEmail, "exists" => 1, "medhubId" => $medhubId]);
-                } else {
-                    Attending::insert(["name" => $name, "email" => $osuEmail, "exists" => 1, "id" => $medhubId]);
-                }
-
-                // echo $userType." ".$name.' added to database'."\n";
-                $added = true;
-            } else {
-                //not an an Anesthesiology resident
-                $emailMessage =
-                    $emailMessage .
-                    $userType .
-                    " " .
-                    $name .
-                    " was found by OSU Find People but is not an Anesthesiology " .
-                    $userType .
-                    ". Please check the information and add user to database manually.";
-                self::notifyAddUser($userType, "David", "david.stahl@osumc.edu", $name, $emailMessage);
-            }
-        } elseif (sizeof($osuMatches) == 0) {
-            // echo '0 residents found in FindPeopleOSU'."\n";
-            $emailMessage =
-                $emailMessage .
-                "No matches for " .
-                $userType .
-                " " .
-                $name .
-                " were found by OSU Find People. The " .
-                $userType .
-                " may be using a preffered name at OSU. Please check the information and add user to database manually.";
-            self::notifyAddUser($userType, "David", "david.stahl@osumc.edu", $name, $emailMessage);
-        } else {
-            // echo 'More than 1 residents found in FindPeopleOSU'."\n";
-            $emailMessage =
-                $emailMessage .
-                "Multiple matches for " .
-                $userType .
-                " " .
-                $name .
-                " were found by OSU Find People. Please check the information and add user to database manually. ";
-            self::notifyAddUser($userType, "David", "david.stahl@osumc.edu", $name, $emailMessage);
-        }
-        // echo var_dump($osuMatches);
-        return $added;
-    }
-
-    // Get user's medhubId from Medhub. Get user's name.# email address from OSU Find People.
-    // $userType is eiterh 'Resident' or 'Attending'
-    public function addUserFromMedhub($userType, $name)
+    // Get Medhub Id using name and Medhub API
+    public function getMedhubId($userType, $name)
     {
-        $userAdded = false;
-        $medhubMatches = null;
+        $medhubId = null;
+        $medhubMatches = array();
         if (strcmp($userType, "Attending") == 0) {
-            $medhubMatches = json_decode(
-                self::medhubPOST("users/facultySearch", json_encode(["name" => $name]))->getBody(),
-                true
-            );
+            try {
+                $medhubMatches = json_decode(self::medhubPOST("users/facultySearch", json_encode(array("name" => $name)))->getBody(), true);
```

Comment:
It seems inconvenient to the caller that `medhubPOST` doesn't handle JSON encoding/decoding.

Reply:
Same comment as before, this PR is not a refactor or MedhubController - I didn't touch anything except updating the method to get the medhub id

Reply:
Makes sense. Do you think this change would be a worthwhile refactoring for another PR?

Reply:
I think that it will go on the backlog and given time constraints (we should finish pushing all our changes to prod next week so we can have a week to fix any issues that might arise) I don't think we will get to it since it is not a high priority, unless you or Spencer think it is?

Reply:
Agreed that deploying our changes successfully is a much higher priority than fixing working code, even if it has a really bad API.